posix/posix.c

```diff
@@ -330,51 +351,57 @@ int posix_clone(int ppid)
 
 	p->process = process_getPid(proc);
 
-	p->fds = vm_kmalloc(p->fdsz * sizeof(fildes_t));
+	p->fds = vm_kmalloc((unsigned int)p->fdsz * sizeof(fildes_t));
 	if (p->fds == NULL) {
-		proc_lockDone(&p->lock);
+		/* MISRAC2012-RULE_17_7-a */
+		(void)proc_lockDone(&p->lock);
 		vm_kfree(p);
 		if (pp != NULL) {
-			proc_lockClear(&pp->lock);
+			/* MISRAC2012-RULE_17_7-a */
+			(void)proc_lockClear(&pp->lock);
 			pinfo_put(pp);
 		}
 		return -ENOMEM;
 	}
 
 	if (pp != NULL) {
-		hal_memcpy(p->fds, pp->fds, pp->fdsz * sizeof(fildes_t));
+		hal_memcpy(p->fds, pp->fds, (unsigned int)pp->fdsz * sizeof(fildes_t));
 
 		for (i = 0; i < p->fdsz; ++i) {
 			f = p->fds[i].file;
 			if (f != NULL) {
-				proc_lockSet(&f->lock);
+				/* MISRAC2012-RULE_17_7-a */
+				(void)proc_lockSet(&f->lock);
 				++f->refs;
-				proc_lockClear(&f->lock);
+				/* MISRAC2012-RULE_17_7-a */
+				(void)proc_lockClear(&f->lock);
 			}
 		}
 
-		proc_lockClear(&pp->lock);
+		/* MISRAC2012-RULE_17_7-a */
+		(void)proc_lockClear(&pp->lock);
 	}
 	else {
-		hal_memset(p->fds, 0, p->fdsz * sizeof(fildes_t));
+		hal_memset(p->fds, 0, (unsigned int)p->fdsz * sizeof(fildes_t));
 
 		for (i = 0; i < 3; ++i) {
 			f = vm_kmalloc(sizeof(open_file_t));
 			p->fds[i].file = f;
 			if (f == NULL) {
-				for (--i; i >= 0; --i) {
-					posix_putUnusedFile(p, i);
+				for (j = 1; i - j >= 0; j++) {
```

Comment:
file #0 was never cleared
```suggestion
				for (j = 0; j < i; j++) {
```